ArcGISRuntimeSDKQt_CppSamples/UtilityNetwork/TraceUtilityNetwork/TraceUtilityNetwork.cpp

```diff
@@ -80,56 +100,51 @@ TraceUtilityNetwork::TraceUtilityNetwork(QObject* parent /* = nullptr */):
 
   // set unique value renderer to the line layer
   m_lineLayer->setRenderer(m_uniqueValueRenderer);
+}
 
-  // Add electric distribution lines and electric devices layers
-  m_map->operationalLayers()->append(m_lineLayer);
-  m_map->operationalLayers()->append(m_deviceLayer);
+void TraceUtilityNetwork::loadUtilityNetwork(const Error& error)
+{
+  if (hasErrorOccurred(error))
+    return;
 
-  connect(m_map, &Map::doneLoading, this, [this](Error e)
-  {
-    if (!e.isEmpty())
-    {
-      m_dialogText = QString(e.message() + " - " + e.additionalMessage());
-      emit dialogVisibleChanged();
-      return;
-    }
+  // Create graphics overlay and append to mapview
+  m_graphicsOverlay = new GraphicsOverlay(this);
+  m_mapView->graphicsOverlays()->append(m_graphicsOverlay);
 
-    m_graphicsOverlay = new GraphicsOverlay(this);
-    m_mapView->graphicsOverlays()->append(m_graphicsOverlay);
+  m_utilityNetwork = new UtilityNetwork(m_serviceUrl, m_map, m_cred, this);
 
-    m_utilityNetwork = new UtilityNetwork(m_serviceUrl, m_map, m_cred, this);
+  connect(m_utilityNetwork, &UtilityNetwork::errorOccurred, this, &TraceUtilityNetwork::hasErrorOccurred);
 
-    connect(m_utilityNetwork, &UtilityNetwork::errorOccurred, [this](Error e)
-    {
-      if (e.isEmpty())
-        return;
+  connect(m_utilityNetwork, &UtilityNetwork::traceCompleted, this, &TraceUtilityNetwork::onTraceCompleted);
 
-      m_dialogText = QString(e.message() + " - " + e.additionalMessage());
-      emit dialogTextChanged();
-    });
+  connect(m_utilityNetwork, &UtilityNetwork::doneLoading, this, &TraceUtilityNetwork::addUtilityNetworkToMap);
 
-    // select the features returned from the trace
-    connect(m_utilityNetwork, &UtilityNetwork::traceCompleted, this, &TraceUtilityNetwork::onTraceCompleted);
+  m_utilityNetwork->load();
+  m_busy = true;
```

Comment:
minor thing - but you could extract `setBusy` as a function that takes care of setting the variable and emitting the signal